exporters/otlp/all/src/main/java/io/opentelemetry/exporter/otlp/internal/OtlpConfigUtil.java

```diff
@@ -160,6 +166,79 @@ public static void configureOtlpExporterBuilder(
     ExporterBuilderUtil.configureExporterMemoryMode(config, setMemoryMode);
   }
 
+  /** Invoke the setters with the OTLP configuration for the {@code dataType}. */
+  @SuppressWarnings("TooManyParameters")
+  public static void configureOtlpExporterBuilder(
+      String dataType,
+      StructuredConfigProperties config,
+      Consumer<String> setEndpoint,
+      BiConsumer<String, String> addHeader,
+      Consumer<String> setCompression,
+      Consumer<Duration> setTimeout,
+      Consumer<byte[]> setTrustedCertificates,
+      BiConsumer<byte[], byte[]> setClientTls,
+      Consumer<RetryPolicy> setRetryPolicy,
+      Consumer<MemoryMode> setMemoryMode) {
+    String protocol = getStructuredConfigOtlpProtocol(config);
+    boolean isHttpProtobuf = protocol.equals(PROTOCOL_HTTP_PROTOBUF);
+    URL endpoint = validateEndpoint(config.getString("endpoint"), isHttpProtobuf);
+    if (endpoint != null && isHttpProtobuf) {
+      String path = endpoint.getPath();
+      if (!path.endsWith("/")) {
+        path += "/";
+      }
+      path += signalPath(dataType);
+      endpoint = createUrl(endpoint, path);
+    }
+    if (endpoint != null) {
+      setEndpoint.accept(endpoint.toString());
+    }
+
+    StructuredConfigProperties headers = config.getStructured("headers");
+    if (headers != null) {
+      headers
+          .getPropertyKeys()
+          .forEach(
+              header -> {
+                String value = headers.getString(header);
+                if (value != null) {
+                  addHeader.accept(header, value);
+                }
+              });
+    }
+
+    String compression = config.getString("compression");
+    if (compression != null) {
+      setCompression.accept(compression);
+    }
+
+    Integer timeoutMs = config.getInt("timeout");
+    if (timeoutMs != null) {
+      setTimeout.accept(Duration.ofMillis(timeoutMs));
+    }
+
+    String certificatePath = config.getString("certificate");
+    String clientKeyPath = config.getString("client_key");
+    String clientKeyChainPath = config.getString("client_certificate");
+
+    if (clientKeyPath != null && clientKeyChainPath == null) {
+      throw new ConfigurationException("client_key provided but client_certificate");
```

Comment:
this message is a little cryptic. Should it be "client_key provided but no client_certificate was configured" ?